animation.js

```diff
@@ -85,15 +89,128 @@ Animation.prototype.update = function () {
       const interpolationValue = (this.time - prevInput) / (nextInput - prevInput)
 
       let currentOutput = null
-      // TODO: stop creating new arrays every frame
-      if (path === 'rotation') {
-        currentOutput = quat.copy(prevOutput)
-        quat.slerp(currentOutput, nextOutput, interpolationValue)
-      } else {
-        currentOutput = []
-        for (var k = 0; k < nextOutput.length; k++) {
-          currentOutput[k] = prevOutput[k] + interpolationValue * (nextOutput[k] - prevOutput[k])
-        }
+      
+      switch(interpolationType){
+        case 'STEP':
+          currentOutput = outputData[prevIndex]
+          break;
+        case 'CUBICSPLINE':
+          //intangent = index
+          //position = index+1
+          //outtangent = index+2
+              
+          let prevInTangent,prevOutTangent,prevPosition,nextInTangent,nextOutTangent,nextPos;
+
+          if(path === 'rotation') {
```

Comment:
Can you make a ternary of this? Knowing that rotation is the only specific case eg `let prevInTangent  = path === 'rotation' ? vec4.create() : vec3.create()`

Reply:
Actually the ternary could happen below: `let prevOutTangent = outputData[(prevIndex * 3) +2].slice()  : [0, 0, 0]`.

Also quick style comment: add space before and after `*` and `+`